### PR TITLE
[usecase] Add Set-Cookie test

### PR DESCRIPTION
**Pull request by chunywang**
The XMLHttpRequest API need docroot for usecase in tinyweb, it is blocked by
https://crosswalk-project.org/jira/browse/XWALK-5905.

Impacted tests(approved): new 1, update 0, delete 0
Unit test platform: Crosswalk Project for Android 18.46.455.0
Unit test result summary: pass 1, fail 0, block 0

BUG=https://crosswalk-project.org/jira/browse/XWALK-5985

### CODE REVIEW

File: usecase/usecase-webapi-xwalk-tests/samples/SetCookie/res/headers.cgi [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+#ARE DISCLAIMED. IN NO EVENT SHALL INTEL CORPORATION BE LIABLE FOR ANY DIRECT,
+#INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+#BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+#DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+#OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+#NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+#EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#Authors:
+#        Wang, Jing J <jing.j.wang@intel.com>
```
Comment (zqzhang) [inline usecase/usecase-webapi-xwalk-tests/samples/SetCookie/res/headers.cgi]: Are you copying this file from somewhere?

File: usecase/usecase-webapi-xwalk-tests/samples/SetCookie/index.html [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+ARE DISCLAIMED. IN NO EVENT SHALL INTEL CORPORATION BE LIABLE FOR ANY DIRECT,
+INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+Authors:
+        Wang, Chunyan <chunyanx.wang@intel.com>
```
Comment (zqzhang) [inline usecase/usecase-webapi-xwalk-tests/samples/SetCookie/index.html]: If possible, prefer the author name in a head `<link>`.

File: usecase/usecase-webapi-xwalk-tests/samples/SetCookie/REAMDE.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+## Usecase Design
+
+This sample demonstrates HTTP response header Set-Cookie.
+
+This usecase covers following interface and methods:
+
+* XMLHttpRequest interface: open send getResponseHeader onreadystatechange
```
Comment (zqzhang) [inline usecase/usecase-webapi-xwalk-tests/samples/SetCookie/REAMDE.md]: Can we add a link to JIRA feature?

```
## Feature Requirement

<link to the JIRA feature>
```

File: usecase/usecase-webapi-xwalk-tests/samples/SetCookie/index.html [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+<script src="../../js/jquery-2.1.3.min.js"></script>
+<script src="../../js/bootstrap.min.js"></script>
+<script src="../../js/common.js"></script>
+<script src="../../js/tests.js"></script>
+<div id="header">
+  <h3 id="main_page_title"></h3>
+</div>
+<div class="content">
+  <iframe id="ifr" frameborder="no" height="400"
+          src="http://127.0.0.1:8081/opt/usecase-webapi-xwalk-tests/samples/SetCookie/res/iframe.html">
```
Comment (zqzhang) [inline usecase/usecase-webapi-xwalk-tests/samples/SetCookie/index.html]: Is this local host "http://127.0.0.1:8081/" supported by default?

File: usecase/usecase-webapi-xwalk-tests/steps/SetCookie/step.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+var step = '<font class="fontSize">'
+         + '<p>Test Purpose: </p>'
+         + '<p>Verifies HTTP response header "Set-Cookie".</p>'
+         + '<p>Test steps: </p>'
+         + '<p>Check the "Set-Cookie" value.</p>'
+         + '<p>Expected Result: </p>'
+         + '<p>Test passes if it should get the response header value expect "Set-Cookie", the value of "Set-Cookie" should be blank(null).</p>'
```
Comment (zqzhang) [inline usecase/usecase-webapi-xwalk-tests/steps/SetCookie/step.js]: Actually in short the expected result is "Set-Cookie response header value is blank(null)", right?

Tester may confuse here. Does the "blank(null)" means nothing or a string? Why the Set-Cookie value is so strange?

https://en.wikipedia.org/wiki/HTTP_cookie#Setting_a_cookie has an example like this

```
HTTP/1.0 200 OK
Content-type: text/html
Set-Cookie: theme=light
Set-Cookie: sessionToken=abc123; Expires=Wed, 09 Jun 2021 10:18:14 GMT
...
```

Can we make the value something human readable and meaningful?

File: usecase/usecase-webapi-xwalk-tests/samples/SetCookie/index.html (hunk 2 above)
Comment (chunywang) [inline usecase/usecase-webapi-xwalk-tests/samples/SetCookie/index.html]: yes, i copy the file from tct-xmlhttprequest-w3c-tests suite. OK, i will update it.

File: usecase/usecase-webapi-xwalk-tests/samples/SetCookie/index.html (hunk 4 above)
Comment (chunywang) [inline usecase/usecase-webapi-xwalk-tests/samples/SetCookie/index.html]: no, it doesn't support, so i add the commit it is blocked by https://crosswalk-project.org/jira/browse/XWALK-5905.

File: usecase/usecase-webapi-xwalk-tests/steps/SetCookie/step.js (hunk 5 above)
Comment (chunywang) [inline usecase/usecase-webapi-xwalk-tests/steps/SetCookie/step.js]: Refer to http://www.w3.org/TR/XMLHttpRequest/#the-getresponseheader()-method:
<code>
  If header is a case-insensitive match for Set-Cookie or Set-Cookie2, return null.
</code>

File: usecase/usecase-webapi-xwalk-tests/samples/SetCookie/index.html (hunk 4 above)
Comment (zqzhang) [inline usecase/usecase-webapi-xwalk-tests/samples/SetCookie/index.html]: Seems the test suite shall be BLOCKED; but it gets PASS in commit message.

File: usecase/usecase-webapi-xwalk-tests/samples/SetCookie/res/iframe.html [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+-->
+
+<meta charset="utf-8">
+<title>Reference File</title>
+<link rel="author" title="Intel" href="http://www.intel.com">
+<link rel="author" title="Chunyan Wang" href="mailto:chunyanx.wang@intel.com">
+<body onload="oninit()">
+  <div>
```
Comment (zqzhang) [inline usecase/usecase-webapi-xwalk-tests/samples/SetCookie/res/iframe.html]: How about changing this code block to make the outputs in alphabetically order?

```
  <ul>
    <li>Content-Type: <span id="ctype"></span></li>
    <li>Response text: <span id="response"></span></li>
    <li>Server: <span id="server"></span></li>
    <li>Set-Cookie: <span id="cookie"></span></li>
    <li>X-Custom-Header: <span id="cusheader"></span></li>
    <li>X-Custom-Header-Comma: <span id="headercomma"></span></li>
  </ul>
```

File: usecase/usecase-webapi-xwalk-tests/samples/SetCookie/index.html (hunk 4 above)
Comment (chunywang) [inline usecase/usecase-webapi-xwalk-tests/samples/SetCookie/index.html]: when i test the case, i add the resource files to any all-in-one test suit(eg:webapi/tct-xmlhttprequest-w3c-test), and change the path to it.

**Pull request merged**